packages/core/test/allOf_test.js

```diff
@@ -30,12 +31,44 @@ describe("allOf", () => {
     expect(node.querySelectorAll("input")).to.have.length.of(1);
   });
 
+  it("should properly merge multiple schemas as per https://json-schema.org/understanding-json-schema/reference/combining.html", () => {
+    const schema = {
+      definitions: {
+        address: {
+          type: "object",
+          properties: {
+            street_address: { type: "string" },
+            city: { type: "string" },
+            state: { type: "string" },
+          },
+          required: ["street_address", "city", "state"],
+        },
+      },
+
+      allOf: [
+        { $ref: "#/definitions/address" },
+        {
+          properties: {
+            type: { enum: ["residential", "business"] },
+          },
+        },
+      ],
+    };
+
+    const { node } = createFormComponent({
+      schema,
+    });
+
+    expect(node.querySelectorAll("input")).to.have.length.of(3); // Schema 1
+    expect(node.querySelectorAll("select")).to.have.length.of(1); // Schema 2
+  });
+
   it("should be able to handle incompatible types and not crash", () => {
     const schema = {
       type: "object",
       properties: {
         foo: {
-          allOf: [{ type: "string" }, { type: "boolean" }],
+          allOf: [{ type: "string" }, { type: "boolean" }], // this will basically pick up the last entry
```

Comment:
What's the justification for picking up the last entry here? Isn't it better to show a validation error because technically the form data will never validate?